### ISSUE TITLE
Picture in picture functionality in iOS 9.0+ devices

### ISSUE THREAD

**Issue by minoue10** [youtube/youtube-ios-player-helper#211]
TL;DR: iOS 9's new `allowsPictureInPictureMediaPlayback` field in `UIWebView` does not play nicely with the framework on iPad.

On my iPad iOS 9.0+, using this library to watch videos reveals the the picture in picture view that Apple released using `allowsPictureInPictureMediaPlayback` (Apple Docs [here](https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIWebView_Class/#//apple_ref/occ/instp/UIWebView/allowsPictureInPictureMediaPlayback)). I believe this change came widely from WebKit (similar field [here](https://developer.apple.com/library/ios/documentation/WebKit/Reference/WKWebViewConfiguration_Ref/index.html#//apple_ref/occ/instp/WKWebViewConfiguration/allowsPictureInPictureMediaPlayback)), but was introduced into UIKit via `UIWebView` for consistency, possibly.

afaik there isn't much support for this pip functionality through `UIWebView`/`WKWebView` (although there are ways to fully take advantage of pip not using WebKit with the `AVPlayerViewController` class, docs [here](https://developer.apple.com/library/ios/documentation/WindowsViews/Conceptual/AdoptingMultitaskingOniPad/QuickStartForPictureInPicture.html)), but since this library uses `UIWebView` it's hard to get any callbacks on any user actions when interacting with the pip. For example, when a user is watching a video that is displayed in pip mode, and when a user taps on the button in the pip to go back to full screen mode, there aren't any callbacks to the `UIWebView` that this has happened. 

It looks like the code that handles the pip functionality in WebKit is in the [WebCore code](https://github.com/WebKit/webkit/blob/master/Source/WebCore/platform/ios/WebVideoFullscreenInterfaceAVKit.mm), which is not much of a help. 

I'm hoping either A) there's actually a way to interact with the Picture in Picture functionality via WebKit that Apple has provided that I just haven't been able to find or B) `allowsPictureInPictureMediaPlayback` will be exposed in this library to have the option to turn it off. 

I'll submit a change for B, but if anyone can provide information about A that would be most helpful.

**Comment by minoue10** [comment]
So it looks like Apple's WebKit references a presentationMode property that is used to determine which mode the video is being played in (read [here](https://developer.apple.com/reference/webkitjs/htmlvideoelement))
as referenced in a [WWDC 2015 video](https://developer.apple.com/videos/play/wwdc2015/501/?time=1903) for Safari and WebKit.

This makes me think it's something that the actual youtube js library needs to concern itself with. Nonetheless, it would be great to turn off PIP until this is done.

**Comment by minoue10** [comment]
Some more references: Apple's documentation on [Picture in Picture in iOS 9](https://developer.apple.com/library/prerelease/content/releasenotes/General/WhatsNewInSafari/Articles/Safari_9_0.html#//apple_ref/doc/uid/TP40014305-CH9-SW18)

**Comment by minoue10** [comment]
Btw - for anyone looking - you can set the `YTPlayerView's` webview property's `allowsPictureInPictureMediaPlayback` to false to turn off picture in picture without having to touch the framework's code. It would still be nice for this library to support picture in picture fully, especially since the Youtube app's pip feature is really nice.

**Comment by alphastryk** [comment]
@minoue10 Setting YTPlayerView''s webView's allowsPictureInPictureMediaPlayback doesn't seem to be removing the PiP button and functionality for me (using the version in CocoaPods, 0.1.4). Was there anything else you had to do to remove that functionality?

**Comment by Jpunt** [comment]
I'm not totally sure, but on my end this Picture in Picture feature seems to be working great. The only glitch I can notice is that it doesn't automatically closes the PIP-window as soon as the view where te player is rendered gets removed (intentionally). Is this something we (/you) can have control over?

**Comment by mac-cain13** [comment]
I've tested this together with @Jpunt in our app, we got quite far supporting PiP, but in the end we keept getting freezes where the app wasn't responding after destroying the `YTPlayerView`. So long story short we also went with setting the `allowsPictureInPictureMediaPlayback` to `false`.
